library-system/src/components/LoginForm.js

```diff
@@ -0,0 +1,42 @@
+// src/components/LoginForm.js
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import '../style.css';
+
+const LoginForm = ({ onSignIn }) => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const navigate = useNavigate();
+
+  const handleSignIn = (e) => {
+    e.preventDefault();
+    
+    if (
+        username.length === 5 &&
+        password.match(/^(?=.*[A-Za-z])(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8}$/)
```

Comment:
password requirements were not matching, number should not be required, and it should not exceed 8 chars